### ISSUE TITLE
AbstractRNNCell documentation

### ISSUE THREAD

**Issue by AlecBG** [tensorflow/tensorflow#40415]
# Documentation for state in AbstractRNNCell could be more clear.

In the documentation for the `AbstractRNNCell` it does not make it clear that the state is a tuple. https://www.tensorflow.org/api_docs/python/tf/keras/layers/AbstractRNNCell

This was a gotcha for me when I defined a custom RNN cell that had a single state. It kept adding an axis to that state whenever I performed an operation on it.

For example, the code within the call method the class implementing `AbstractRNNCell`
```logging.info(f'states: {states}')
logging.info(f'state_update: {state_update}')
new_states = tf.math.add(states, state_update)
logging.info(f'new_states: {new_states}')
```

leads to the confusing output

```06-12 12:28 root         INFO     states: (<tf.Tensor 'Placeholder_3:0' shape=(32, 4) dtype=float32>,)
06-12 12:28 root         INFO     state_update: Tensor("add_1:0", shape=(32, 4), dtype=float32)
06-12 12:28 root         INFO     new_states: Tensor("Add_2:0", shape=(1, 32, 4), dtype=float32)
```

Upon implementing the state as a tuple of length one, the issue was solved. I think this could be made more clear in the documentation.

Many thanks.

**Comment by jvishnuvardhan** [comment]
@AlecBG Is this still an issue for you? The updated doc clearly mentions about `state_size`



state_size | size(s) of state(s) used by this cell.It can be represented by an Integer, a TensorShape or a tuple of Integers or TensorShapes.
-- | --




https://www.tensorflow.org/api_docs/python/tf/keras/layers/AbstractRNNCell

Thanks!
